File: android/src/main/java/de/homeinfogmbh/plugins/ble/CapBeacon.java
package de.homeinfogmbh.plugins.ble;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import java.util.List;
import java.util.Objects;

import org.altbeacon.beacon.Beacon;
import org.altbeacon.beacon.Identifier;

public class CapBeacon {

  double distance;
  int serviceUuid;
  Identifier id1;
  Identifier id2;
  Identifier id3;
  List<Long> dataFields;
  int rssi;
  int txPower;
  String bluetoothAddress;
  String bluetoothName;


    CapBeacon(@NonNull Beacon beacon) {
        this.distance = beacon.getDistance();
        this.serviceUuid = beacon.getServiceUuid();
        this.id1 = beacon.getId1();
        this.id2 = beacon.getId2();
        this.id3 = beacon.getId3();
        this.dataFields = beacon.getDataFields();
        this.rssi = beacon.getRssi();
        this.txPower = beacon.getTxPower();
        this.bluetoothAddress = beacon.getBluetoothAddress();
        this.bluetoothName = beacon.getBluetoothName();
    }

    @Override
  public boolean equals(Object o){
    if (this == o) return true;
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    CapBeacon newBeacon = (CapBeacon) o;
    return (
      distance >= ((CapBeacon) o).distance*0.8 &&
      distance <= ((CapBeacon) o).distance*1.2 &&
      serviceUuid == newBeacon.serviceUuid &&
      id1 == newBeacon.id1 && id2 == newBeacon.id2 &&
      id3 == newBeacon.id3 && Objects.equals(bluetoothAddress, newBeacon.bluetoothAddress) &&
      Objects.equals(bluetoothName, newBeacon.bluetoothName)
    );

    }
}

File: ios/Plugin/CapBeacon.swift
//
//  CapBeacon.swift
//  Bleindoorlocation
//
//  Created by sebastian on 24.03.23.
//

import Foundation
import CoreLocation

public class CAPBeacon: Encodable {
    var distance: Double
    var serviceUuid: Int
    var id1: String
    var id2: String
    var id3: String
    var dataFields: [UInt64]
    var rssi: Int
    var txPower: Int
    var bluetoothAddress: String
    var bluetoothName: String


    init(beacon: CLBeacon!) {
        self.distance = beacon.accuracy
        //not supported by ios
        self.serviceUuid = 0;
        self.id1 = beacon.uuid.uuidString
        self.id2 = beacon.major.stringValue
        self.id3 = beacon.minor.stringValue
        //not supported by ios
        self.dataFields = []
        self.rssi = beacon.rssi
        //not supported by ios
        self.txPower = 0;
        //not supported by ios
        self.bluetoothAddress = ""
        //not supported by ios
        self.bluetoothName = ""
    }
    
    public func equals(o: CAPBeacon?)-> Bool{
        if (o == nil) {return false}
        if (self === o) {return true};
        return (
            distance >= (o!.distance*0.8) &&
            distance <= (o!.distance*1.2) &&
            id1 == o!.id1 && id2 == o!.id2 &&
            id3 == o!.id3 && bluetoothAddress == o!.bluetoothAddress &&
            bluetoothName == o!.bluetoothName
        );
    
    }
}
